src/org/labkey/test/util/TestDataGenerator.java

```diff
@@ -328,6 +330,31 @@ public static String randomString(int size, @Nullable String exclusion, @Nullabl
         return val.toString();
     }
 
+    public static String randomDomainName()
+    {
+        return randomDomainName(10);
+    }
+
+    public static String randomInvalidDomainName(int size)
+    {
+        return randomString(size, null, ILLEGAL_DOMAIN_NAME_CHARSET);
```

Comment:
Might be better (slightly more realistic) if this consisted of some number of illegal characters along with some legal characters. But perhaps the usages of this method already accomplish that.